src/generate.rs

```diff
@@ -950,7 +950,8 @@ pub fn fields(
             if let Some(write_constraint) = f.write_constraint {
                 match *write_constraint {
                     WriteConstraint::Range(ref range) => {
-                        if range.min == 0 && range.max == (1 << f.width) - 1 {
+                        if range.min as u64 == 0 &&
+                           range.max as u64 == (1 << f.width as u64) - 1 {
```

Comment:
Did you mean `1u64 << f.width`? I don't think is necessary to cast the RHS to u64 since the shift operators works with any integer type on the RHS.